tests/unit/backend/corpora/common/entities/test_dataset.py

```diff
@@ -28,3 +35,91 @@ def test__get__ok(self):
     def test__get__does_not_exist(self):
         non_existent_key = "non_existent_id"
         self.assertEqual(Dataset.get(non_existent_key), None)
+
+    def test__create__ok(self):
+        """
+        Create a dataset with a variable number of artifacts, contributors, and deployment_directories
+
+        """
+        artifact_params = dict(
+            filename="filename_1",
+            filetype=DatasetArtifactFileType.H5AD,
+            type=DatasetArtifactType.ORIGINAL,
+            user_submitted=True,
+            s3_uri="some_uri",
+        )
+
+        deployment_directory_params = dict(environment="test", url="test_url")
+
+        contributor_params = dict(name="bob", institution="school", email="some@email.com")
+
+        dataset_params = dict(
+            name="create_dataset_id",
+            organism="organism",
+            organism_ontology="123",
+            tissue="tissue",
+            tissue_ontology="123",
+            assay="assay",
+            assay_ontology="123",
+            disease="diseas",
+            disease_ontology="123",
+            sex="F",
+            ethnicity="ethnicity",
+            ethnicity_ontology="123",
+            source_data_location="location",
+            preprint_doi="preprint",
+            publication_doi="publication",
+        )
+
+        for i in range(3):
+            with self.subTest(i):
+                dataset = Dataset.create(
+                    **dataset_params,
+                    artifacts=[artifact_params] * i,
+                    contributors=[contributor_params] * i,
+                    deployment_directories=[deployment_directory_params] * i,
+                )
+
+                dataset_id = dataset.id
+                artifact_ids = get_ids(dataset.artifacts)
+                deployment_directory_ids = get_ids(dataset.deployment_directories)
+                contributor_ids = get_ids(dataset.contributors)
+
+                # Expire all local object and retireve them from the DB to make sure the transactions went through.
+                Dataset.db.session.expire_all()
+
+                dataset = Dataset.get(dataset_id)
+                self.assertIsNotNone(dataset)
+                self.assertEqual(dataset_id, dataset.id)
+                self.assertEqual(artifact_ids, get_ids(dataset.artifacts))
+                self.assertEqual(deployment_directory_ids, get_ids(dataset.deployment_directories))
+                self.assertEqual(contributor_ids, get_ids(dataset.contributors))
+
+        with self.subTest("With existing rows"):
```

Comment:
This warrants its own test. It's good practice to keep tests as separate as possible even if there is overlap.

Reply:
moved to its own test